src/main/java/com/yahoo/bullet/query/Query.java

```diff
@@ -32,13 +32,18 @@
     private final Expression filter;
     private final Aggregation aggregation;
     private final List<PostAggregation> postAggregations;
+    private final Query postQuery;
     private Window window;
     private Long duration;
 
     private static final BulletException ONLY_RAW_RECORD = new BulletException("Only RAW aggregation type can have window emit type RECORD.",
                                                                                "Change your aggregation type or your window emit type to TIME.");
     private static final BulletException NO_RAW_ALL = new BulletException("RAW aggregation type cannot have window include type ALL.",
-                                                                          "Change your aggregation type or your window include type");
+                                                                          "Change your aggregation type or your window include type.");
+    private static final BulletException NO_POST_QUERY_WINDOW = new BulletException("Post query cannot have a window.",
```

Comment:
Let's call it Subquery everywhere

Reply:
which one do you prefer?

Reply:
Outer because
1. Bql calls it so
2. It's confusing because the subquery that people think it'd mean would be the inner query but for us it's actually the outermost query